keps/sig-cli/kustomize-default-config.md

```diff
@@ -0,0 +1,314 @@
+in order to skip a transformation for a specific resource or resource kind.
+
+We are aiming at addressing those issues:
+- Make cluster level kind configurable [#617](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Do not add namespace to cluster scoped CRDs [#552](https://github.com/kubernetes-sigs/kustomize/issues/552) 
+- Skipping nameprefix for some resource kind [#519](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Adding labels only to metadata.labels [#330](https://github.com/kubernetes-sigs/kustomize/issues/330)
+- Skip adding labels to certain paths [#519](https://github.com/kubernetes-sigs/kustomize/issues/519)
+- Unable to disable commonLabels injection using transformer config: Error: conflicting fieldspecs [#817](https://github.com/kubernetes-sigs/kustomize/issues/817)
+
+## Motivation
+
+
+### Goals
+
+- Provide the ability for the user to skip transformation for specific group/version/kind especially usefull if the default
+  configuration is using a wild card (see namespace and prefixsuffix transformer configuration).
+- Provide the ability of the user to replace or remove an entry in the default configuration which is not matching his needs
+  (see commonLabel transformer configuration)
+
+  
+### Non-Goals
+
+- Provide backward compatibility with the transformer default configurations.
+- Provide a way for the user to exclude a kind from a wild card based set as opposed to have to list all the possible kinds/gvks.
+- Provide a way for the user to remove a configuration entry from the default configuration if it is not aligned with the needs
+  of his project.
+- Provide a simple method to the transformer developers to select the list of "mutable" fields based on the gvk of the resources.
+
+## Proposal
+
+The propsal has three parts:
+
+1. Update the FieldSpec definition to add a "skip" field as follow (It is also useful external transformers needing wild card):
+
+```go
+type FieldSpec struct {
+	gvk.Gvk            `json:",inline,omitempty" yaml:",inline,omitempty"`
+	Path               string `json:"path,omitempty" yaml:"path,omitempty"`
+	CreateIfNotPresent bool   `json:"create,omitempty" yaml:"create,omitempty"`
+	SkipTransformation bool   `json:"skip,omitempty" yaml:"skip,omitempty"`
+}
+```
+
+   
+2. Wrap []FieldSpec into FieldSpecs and add high level method to ease implementation of transformers, during loop accross
+   resources and fieldspec.
+
+```go
+type FieldSpecs []FieldSpec
+
+func (s FieldSpecs) ApplicableFieldSpecs(x gvk.Gvk) FieldSpecs {
+// ....
+}
+```
+
+3. Wrap FieldSpec into a FieldSpecConfig, to let the user remove and replace configuration entries embedded in the default configuration
+   provided with kustomize
+   
+```go
+type FieldSpecConfig struct {
+	FieldSpec `json:",inline,omitempty" yaml:",inline,omitempty"`
+ // Behavior legal values are "", "add", "replace", "remove"
+	Behavior  string `json:"behavior,omitempty" yaml:"behavior,omitempty"`
```

Comment:
Why you need this. The `skip` field from 1 can already overwrite the default behavior.